lightning-rapid-gossip-sync/src/processing.rs

```diff
@@ -29,10 +32,30 @@ const GOSSIP_PREFIX: [u8; 4] = [76, 68, 75, 1];
 /// avoid malicious updates being able to trigger excessive memory allocation.
 const MAX_INITIAL_NODE_ID_VECTOR_CAPACITY: u32 = 50_000;
 
+/// We remove disallow gossip data that's more than two weeks old, per BOLT 7's
+/// suggestion.
+const STALE_RGS_UPDATE_AGE_LIMIT_SECS: u64 = 60 * 60 * 24 * 14;
+
 impl<NG: Deref<Target=NetworkGraph<L>>, L: Deref> RapidGossipSync<NG, L> where L::Target: Logger {
 	pub(crate) fn update_network_graph_from_byte_stream<R: io::Read>(
+		&self,
+		read_cursor: &mut R,
+	) -> Result<u32, GraphSyncError> {
+		#[allow(unused_mut)]
+		let mut current_time_unix = None;
+		#[cfg(all(feature = "std", not(test)))]
+		{
+			// Note that many tests rely on being able to set arbitrarily old timestamps, thus we
+			// disable this check during tests!
```

Comment:
As of now, the test inputs are a bit mysterious. If we programmatically construct them rather than use fixed values, shouldn't we be able to enable this in testing (and make the test more understandable)?

Reply:
I agree. See my other comment :)